src/common/libsubprocess/remote.c

```diff
@@ -615,7 +606,21 @@ static void rexec_continuation (flux_future_t *f, void *arg)
 
     if (subprocess_rexec_get (f) < 0) {
         if (errno == ENODATA) {
-            remote_completion (p);
+            p->remote_completed = true;
+            /* Per RFC42, when remote processes are launched, the
+             * process should return finished (i.e. state EXITED)
+             * before returning ENODATA.  It is otherwise considered a
+             * protocol error.
+             *
+             * N.B. There is evidence that the sdexec module has
+             * violated the protocol before #6012.
```

Comment:
I would say the more appropriate ref would be #5956 as that's where the larger issue starts.